Godeps/_workspace/src/github.com/evanphx/json-patch/patch.go

```diff
@@ -314,7 +325,38 @@ func (d *partialArray) set(key string, val *lazyNode) error {
 	}
 
 	idx, err := strconv.Atoi(key)
+	if err != nil {
+		return err
+	}
+
+	sz := len(*d)
+	if idx+1 > sz {
+		sz = idx + 1
+	}
+
+	ary := make([]*lazyNode, sz)
+
+	cur := *d
+
+	copy(ary, cur)
+
+	if idx >= len(ary) {
+		fmt.Printf("huh?: %#v[%d] %s, %s\n", ary, idx)
```

Comment:
No idea why - upstream library.... Skype never be hit hithough otherwise will panic on following lines. Worth updating to return understandable error I guess, but should never be reached.